api_tests/preprints/views/test_preprint_versions.py

```diff
@@ -154,3 +153,85 @@ def test_permission_error(self):
         self.post_mod_preprint.add_contributor(user_write, permissions.WRITE)
         res = self.app.post_json_api(self.post_mod_version_create_url, auth=user_write.auth, expect_errors=True)
         assert res.status_code == 403
+
+
+class TestPreprintVersionList(ApiTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user = AuthUserFactory()
+        self.moderator = AuthUserFactory()
+        self.provider = PreprintProviderFactory(_id='osf', name='osfprovider')
+        self.post_mod_preprint = PreprintFactory(
+            reviews_workflow='post-moderation',
+            is_published=True,
+            creator=self.user,
+            provider=self.provider
+        )
+        self.pre_mod_preprint = PreprintFactory(
+            reviews_workflow='pre-moderation',
+            is_published=False,
+            creator=self.user,
+            provider=self.provider
+        )
+        self.pre_mod_preprint.provider.get_group('moderator').user_set.add(self.moderator)
+        self.post_mod_preprint.provider.get_group('moderator').user_set.add(self.moderator)
+        self.latest_preprint = self.post_mod_preprint
+        self.origin_guid = str(self.post_mod_preprint.guids.first()._id)
+        for _ in range(5):
+            new_version = PreprintFactory.create_version(
+                create_from=self.latest_preprint,
+                creator=self.user,
+                set_doi=False
+            )
+            self.latest_preprint = new_version
```

Comment:
Make sure we update preprint states and review states for each version so test fixtures are consistent with how things actually work.

Reply:
This comment still stands.